x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -101,7 +104,29 @@ export const xyChartRenderer: RenderFunction<XYChartProps> = {
   },
 };
 
+function getIconForSeriesType(seriesType: SeriesType): IconType {
+  return chartTypeIcons.find(chartTypeIcon => chartTypeIcon.id === seriesType)!.iconType;
+}
+
 export function XYChart({ data, args }: XYChartProps) {
+  if (data.rows.length === 0) {
```

Comment:
This logic is about to change with layers, especially if we change the type of the Visualization expression args. But this makes sense.